bin/plot-windowed-identity.py

```diff
@@ -0,0 +1,394 @@
+            if thisIdentity >= tolerance * bestIdentity:
+                bestMatches.append((thisIdentity, readId))
+            else:
+                break
+
+        center = centers[jumpIndex]
+        x = [center - half, center + half]
+
+        # Collect the best matches.
+        lastIdentity = 2.0  # Must be > 1.0 to start things off.
+        for thisIdentity, readId in bestMatches:
+            bestCounts[readId] += 1
+            showlegend = readId not in inLegend
+            inLegend.add(readId)
+            if thisIdentity >= lastIdentity:
+                adjustedIdentity = lastIdentity - delta
+            else:
+                adjustedIdentity = thisIdentity
+            lastIdentity = adjustedIdentity
+            try:
+                color = colors[readId]
+            except KeyError:
+                color = colors[readId] = next(colorIterator)
+            scatters[readId].append(
+                go.Scatter(
+                    x=x,
+                    y=[adjustedIdentity, adjustedIdentity],
+                    name=readId,
+                    text=f"{readId} {int(x[0])}-{int(x[1])} {thisIdentity:.4f}",
+                    hoverinfo="text",
+                    legendgroup=readId,
+                    showlegend=showlegend,
+                    mode=mode,
+                    marker=marker,
+                    line_color=color,
+                )
+            )
+
+    # Change legend names to include the count of best windows the sequence is in.
+    for readId, theseScatters in scatters.items():
+        for scatter in theseScatters:
+            scatter["name"] = f"{readId} ({bestCounts[readId]})"
+
+    data = []
+
+    # Add the scatters for each read in the order found in identity.
+    for read in identity:
+        data.extend(scatters[read.id])
+
+    return data
+
+
+def getAllLines(
+    centers: list[float], identity: dict[DNARead, list[float]], args: argparse.Namespace
+) -> list[go.Scatter]:
+    """
+    Plot identity lines for all sequences.
+
+    @param centers: The x-axis centers of the genome windows.
+    @param identity: A mapping from sequences (that have been matched against the
+        reference) to a list of float identities (one for each window).
+    @param title: The title of the plot.
+    @param args: The command-line arguments.
+    """
+    data = []
+
+    mode, marker = (
+        ("lines+markers", dict(size=args.markerSize))
+        if args.markerSize  # i.e., not 0 and not None.
+        else ("lines", None)
+    )
+
+    for read, identities in identity.items():
+        data.append(
+            go.Scatter(
+                x=centers,
+                y=identities,
+                mode=mode,
+                marker=marker,
+                name=read.id,
+            )
+        )
+
+    return data
+
+
+def plotPlotly(
+    centers: list[float],
+    identity: dict[DNARead, list[float]],
+    title: str,
+    args: argparse.Namespace,
```

Comment:
I usually do that, mainly for two reasons: calling external functions that I've imported (and which other people who don't have their hands on a `Namespace` might want to call) and when writing functions that I test. Within a script, with all the args already bundled up into one object that is documented in one place, I sometimes just pass that around because it's convenient and concise. The only real downside I see is that some receiving function might misbehave and access some attribute that it's not meant to. But it's all the one functionality in the one script file, so I'm not worried about that.